File: Csharp/CodeWars/src/Resolutions/HelloNameOrWorld.cs
using System.Globalization;

namespace CodeWars.Resolutions;

public class HelloNameOrWorld
{
    public static string Hello(string name = "") =>
        string.IsNullOrEmpty(name)
            ? "Hello, World!"
            : $"Hello, {CultureInfo.CurrentCulture.TextInfo.ToTitleCase(name.ToLower())}!";
}

File: TypeScript/Codewars/src/resolutions/Hello-Name-or-World.ts
export function hello(name = ""): string {
	const correctName = name === "" ? "World" : analyzeWord(name);

	return `Hello, ${correctName}!`;
}

function analyzeWord(str: string): string {
	return str[0].toUpperCase() + str.slice(1).toLowerCase();
}
